maps/HDRUK/3.0.0/HDRUK/datasetv2/translation.jsonata

```diff
@@ -0,0 +1,419 @@
+    },['publisher']|,
+    "documentation": documentation ~> |$| 
+    {
+      "description": $length(description)=0 ? null : description,
+      "associatedMedia": associatedMedia ~> $to_str ~> $null_string,
+      "inPipeline": $$.extra.inPipeline
+    },['isPartOf']|,
+    "coverage": coverage ~> $cleanAllValues ~> |$|
+    {
+      "materialType": physicalSampleAvailability ~> $safeSplit(";,;") ~> $applyMap($map_materials),
+      "typicalAgeRangeMin": $parseInteger($safeSplit(typicalAgeRange, "-")[0], "#,##0") ~> $defaultValue(null),
+      "typicalAgeRangeMax": $parseInteger($safeSplit(typicalAgeRange, "-")[1], "#,##0") ~> $defaultValue(null)
+    },['physicalSampleAvailability','typicalAgeRange']|,
+    "provenance": {
+      "origin": provenance.origin ~> $cleanAllValues ~> |$|
+      {
+        "purpose": purpose ~> $safeSplit(",") ~> $applyMap($map_purpose),
+        "source": source ~> $safeSplit(",") ~> $applyMap($map_source),
+        "datasetType": $$.extra.datasetType ~> $defaultValue("Health and disease"),
+        "datasetSubType": $$.extra.datasetSubType ~> $defaultValue("Not applicable"),
+        "collectionSource": collectionSituation ~> $safeSplit(";,;") ~> $applyMap($map_collectionSource)
+      },['collectionSituation']|,
+      "temporal": provenance.temporal ~> $cleanAllValues ~> |$|
+         {
+          "publishingFrequency": accrualPeriodicity ~> $applyMap($map_accrualPeriodicity),
+          "distributionReleaseDate": distributionReleaseDate ? distributionReleaseDate ~> $get_date : null,
+          "startDate": startDate ? startDate ~> $get_date : "1970-01-01",
+          "endDate": endDate  ? endDate ~> $get_date : null,
+          "timeLag": timeLag ~> $applyMap($map_deliveryLeadTime) ~> $defaultValue("Not applicable")
+         },['accrualPeriodicity']|
+    },
+    "accessibility": {
+      "usage": accessibility.usage ~> $cleanAllValues ~> |$|
+      {
+        "dataUseLimitation": dataUseLimitation ~> $safeSplit(";,;") ~> $applyMap($map_dataUseLimitations),
+        "dataUseRequirements": dataUseRequirements ~> $safeSplit(";,;") ~> $applyMap($map_dataUseRequirements)
+      },['investigations','isReferencedBy']|,
+      "access": accessibility.access ~> $cleanAllValues ~> |$|
+      {
+        "deliveryLeadTime": deliveryTimeLead ~> $applyMap($map_deliveryLeadTime) ~> $defaultValue("Not applicable"),
+        "accessServiceCategory": $$.extra.accessServiceCategory
+      },['deliveryTimeLead']|,
+      "formatAndStandards": accessibility.formatAndStandards ~> $cleanAllValues ~> |$|
+      {
+        "vocabularyEncodingScheme": vocabularyEncodingScheme ~> $safeSplit(";,;") ~> $checkList($allowedEncodings),
+        "conformsTo": conformsTo ~> $safeSplit(";,;")  ~> $checkList($allowedStandards),
+        "language": language ~> $safeSplit(";,;"),
+        "format": format ~>  $safeSplit(";,;")
+      }|
+    },
+    "enrichmentAndLinkage": enrichmentAndLinkage ~> |$| /*drop linkages*/
```

Comment:
Some linkages in HDRUK v3 can be found in other fields (I think), unless there is a separate plan to migrate these aside from through translation.

```
        "isPartOf": $$.input.documentation.isPartOf,
        "linkableDatasets": $$.input.enrichmentAndLinkage.qualifiedRelation,
        "investigations": $$.input.access.usage.investigations,
        "publicationUsingDataset": $$.input.access.usage.isReferencedBy
```

Reply:
Yeah that's right. These are all redundant and just absolutely full of crap in mk1 (e.g. title names). I took the executive decision to ignore as will be handled through the linkages tables.

Happy if you'd prefer I leave it in to cover us if need be?

Reply:
Fair enough, if the data is messy I agree with your decision that it is better to handle it elsewhere.